src/hooks/useDarkMode.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import { useEffect, useState } from 'react';
+
+export const useDarkMode = (
+  usePreferences: boolean,
+): [Mode, React.Dispatch<React.SetStateAction<Mode>> | null, (() => void) | null] => {
+  if (!usePreferences) return [null, null, null];
+
+  const [mode, setMode] = useState<Mode>(null);
+
+  const toggleMode = () => {
+    if (!mode) return;
+    const newMode = mode == 'light' ? 'dark' : 'light';
+    document.documentElement.className = '';
+    document.documentElement.classList.add(newMode);
```

Comment:
hoho, didn't know about this one.

Reply:
But it will change a lot of the behaviour. :thinking:

Reply:
Yeah, I fact I could change it, but then it will not work as I'm expecting. I'll remove 3 lines here, and add a few more in other places. Don't see it as a :trophy:.

Reply:
I don't see changing these lines as an optimization, I see a bug here that needs to be fixed, because if you do this affectation  `document.documentElement.className = '';` you might be breaking some behavior in the product.